com.unity.perception/Runtime/GroundTruth/PerceptionUpdater.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace UnityEngine.Perception.GroundTruth
+{
+    /// <summary>
+    /// PerceptionUpdater is automatically spawned when the player starts and is used to coordinate and maintain
+    /// static perception lifecycle behaviours.
+    /// </summary>
+    class PerceptionUpdater : MonoBehaviour
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+        static void Initialize()
+        {
+            var updaterObject = new GameObject("PerceptionUpdater");
+            updaterObject.AddComponent<PerceptionUpdater>();
+            updaterObject.hideFlags = HideFlags.HideAndDontSave;
+            DontDestroyOnLoad(updaterObject);
+        }
+
+        void LateUpdate()
+        {
+            LabelManager.singleton.RegisterPendingLabels();
```

Comment:
This would cause undefined behavior with object creation/deletion in other LateUpdate() scripts. To avoid this causing user confusion, we should instead use [SetPlayerLoop](https://docs.unity3d.com/ScriptReference/LowLevel.PlayerLoop.SetPlayerLoop.html) to avoid this entirely.

Reply:
Good point. Another option I've used for the ScenarioBase class is the DefaultExecutionOrder attribute. If I add the following attribute configuration to the LabelManager class, it should always run after all other default MonoBehaviour implemenations:
[DefaultExecutionOrder(1)]

Reply:
This caught my attention too. Let's make sure we don't get undefined behaviour here. Apart from the attribute, we could also use [this](https://docs.unity3d.com/Manual/class-MonoManager.html) window to set the execution order to sth very high.

Reply:
The attribute I suggested is equivalent to manipulating the script's position using this window. As for setting the value very high, I'll make sure to set it reasonably high so that the average user won't author any scripts that will interfere with this updater, but there are other default Unity processes that have low priority values, so I'll probably avoid going past those.